lib/Basics/ResultError.cpp

```diff
@@ -0,0 +1,62 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ResultError.h"
+
+#include "Basics/StaticStrings.h"
+#include "Basics/error.h"
+#include "Basics/voc-errors.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include <ostream>
+
+using namespace arangodb;
+using namespace arangodb::result;
+
+Error::Error(int errorNumber) noexcept(noexcept(std::allocator<char>()))
+    : _errorNumber(errorNumber) {}
+
+Error::Error(int errorNumber, std::string_view const& errorMessage)
+    : _errorNumber(errorNumber), _errorMessage(errorMessage) {}
+
+auto Error::errorNumber() const noexcept -> int { return _errorNumber; }
+
+auto Error::errorMessage() const -> std::string_view {
+  if (!_errorMessage.empty()) {
+    return _errorMessage;
+  }
+  return TRI_errno_string(_errorNumber);
```

Comment:
One could make this return a `string_view` - just as an idea. The length is known at compile time.

Reply:
Agreed, I've also thought about that. Though I tend to do that change in a separate PR.

Reply:
@maierlars https://github.com/arangodb/arangodb/pull/13455